### PR TITLE
Adding Default Resolution Values

### PR DESCRIPTION
**Pull request by DhairyaBahl**
@mhuzai 
This is the PR for the issue #132 .

Kindly review my PR and suggest me further changes if i can do any. 

Thanks :+1:

### CODE REVIEW

**Comment by DhairyaBahl** [comment]
@charmoniumQ @mhuzai Kindly Review my PR so that i can make further changes if required.

**Comment by e3m3** [comment]
Hey @DhairyaBahl! How are things going? Are you still interested in working on this PR? We are looking to finish up this PR within a few days, but I am here to guide you through this PR if you are still up to the task :). If not, don't worry, we can find you another good first issue if you are still interested in contributing to the project.

If you choose to work on this PR, here are some missing things:
- [ ] Rename your new header file so that we indicate that it should be used for other variables! (See `man git-mv`)
- [ ] Rename your variables to have `ALL_CAPITAL_LETTERS` to indicate that they are runtime constants
- [ ] Hologram's `SLM_WIDTH` and `SLM_HEIGHT` should reflect your new default values
- [ ] Add your information to the `CONTRIBUTORS` and `AUTHORS` files per the [contributing guidelines](https://github.com/ILLIXR/ILLIXR/blob/cb1888a54c4cf31647a3ee4b531a38945ff21d39/CONTRIBUTING.md)

(@mhuzai || @charmoniumQ): The branch name for the closed PR #139 was correct, but this one is not. Should we close this one and reopen the other?

**Comment by DhairyaBahl** [comment]
@e3m3 Hello, I was expecting someone to tell me what changes do i have to make in this PR. It was under review !! . I would really like to work in this project and in this issue. So I will make changes in the PR as per your previous message. 

Thanks you :)

**Comment by DhairyaBahl** [comment]
@e3m3 I made some changes and pushed them in this PR. Kindly review it and tell me what more to do. and I am unable to find ```SLM_HEIGHT AND SLM_WIDTH``` anywhere. After I pushed changes from upstream to local, even the hologram folder vanished. 😅. Thanks for help and effort . :)

**Comment by e3m3** [comment]
No worries @DhairyaBahl, thanks for working on this!

We would still like to address the first task I mentioned regarding giving `default_resolution.hpp` a more meaningful name if it is to hold other constants. Something like `global_module_defs.hpp` perhaps? Suggestions welcome!

Indeed, the hologram folder is missing because it is now hosted at this [GitHub fork](https://github.com/ILLIXR/HOTlab). You should clone that repository, make a new branch `issue-132-make-the-default-resolution-2k` in that repository, make your changes, and then start a PR for that project with your changes. Your changes in the new project should look something like this:
```C
#ifdef YOUR_VARIABLE_NAME
    // set the hologram variable here ...
#else
    // keep the old variable definition here ...
#endif
```
Do this for both variables in hologram that you need to handle. Doing this macro-based variable definition allows us to provide backward compatibility for users of ILLIXR who are not using the most up-to-date branch, or users of the forked hologram code that are using it separately from the main module code.

The last step after this is to add the matching `#define ...` in the main project where you defined your variables originally. Doing so allows the `#ifdef ...` in the hologram project to know that the code in the main module is using your changes! This would look as follows for each variable:
```C
#ifndef YOUR_VARIABLE_NAME
#define YOUR_VARIABLE_NAME
    // your variable declaration/definition here ...
#endif
```
This code checks that you haven't defined the variable elsewhere in the code. If you haven't, it will define and run your code for you. See this [article](https://en.wikipedia.org/wiki/C_preprocessor#Conditional_compilation) for more information about conditional compilation with macros.

Let us know if you have more questions about this.

**Comment by DhairyaBahl** [comment]
@e3m3 I have pushed the required changes in this PR. If they are good to go for this repo then I will stash the commits finally. 

Thanks :)

**Comment by DhairyaBahl** [comment]
@e3m3 Did I made the changes correctly in the headerfile. I meant those #define and #indef. Kindly tell me if i did that wrong and also review the PR of the hologram project.

Thank you :)

**Comment by DhairyaBahl** [comment]
@e3m3 Kindly tell me if there are any changes to be made in this. Thank you :)

**Comment by charmoniumQ** [comment]
Looks good to me.

**Review by charmoniumQ — APPROVED** [review]

**Review by e3m3 — DISMISSED** [review]

File: CONTRIBUTORS [hunk 1]
```diff
@@ -26,3 +26,4 @@ Joseph Ravichandran <jpr3@illinois.edu>
 Finn Sinclair <fns2@illinois.edu>
 Hengzhi Yuan <hengzhi2@illinois.edu>
 Jeffrey Zhang <jfzhang2@illinois.edu>
+Dhairya Bahl <dhairyabahl5@gmail.com>
```
Comment (mhuzai) [inline CONTRIBUTORS:29]: Alphabetical order.

File: common/global_module_defs.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+This is the file where default resolution values are defined
+*/
+
+
+//setting default width
+static constexpr int DEFAULT_WIDTH=2560;
+
+//setting default height
+static constexpr int DEFAULT_HEIGHT=1440;
```
Comment (mhuzai) [inline common/global_module_defs.hpp:10]: What are these for?

Comment (DhairyaBahl) [reply common/global_module_defs.hpp:10]: @mhuzai Actually I was using it before for files like gldemo/plugin.cpp coz it was giving error like there must be same datatype static and constexpr to use it in the file. So , i put it there and it worked. Now I added #DEFINE so i guess it should do the task so I will remove it. :)

File: CONTRIBUTORS [hunk 3]
```diff
@@ -21,6 +21,7 @@ Muhammad Huzaifa <huzaifa2@illinois.edu>
 Jae Lee <jael2@illinois.edu>
 Xutao Jiang <xutaoj2@illinois.edu>
 Ying Jing <yingj4@illinois.edu>
+Dhairya Bahl <dhairyabahl5@gmail.com>
```
Comment (e3m3) [inline CONTRIBUTORS:24]: Move to alphabetical order (the second entry)

File: common/global_module_defs.hpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+//setting default width
+#ifndef DEFAULT_WIDTH
+#define DEFAULT_WIDTH 2560
+#endif
+
+//setting default height 
+#ifndef DEFAULT_HEIGHT
+#define DEFAULT_HEIGHT 1440
+#endif
```
Comment (e3m3) [inline common/global_module_defs.hpp:20]:
```suggestion
#pragma once

namespace ILLIXR {

#ifndef FB_WIDTH
#define FB_WIDTH FB_WIDTH
// Setting default framebuffer width
static constexpr int FB_WIDTH = 2560; // Pixels
#endif // FB_WIDTH

#ifndef FB_HEIGHT
#define FB_HEIGHT FB_HEIGHT
// Setting default framebuffer height
static constexpr int FB_HEIGHT = 1440; // Pixels
#endif // FB_HEIGHT

}
```

File: gldemo/plugin.cpp [hunk 5]
```diff
@@ -14,11 +14,12 @@
...
 #include "common/gl_util/obj.hpp"
 #include "shaders/demo_shader.hpp"
+#include "common/global_module_defs.hpp"
 
 using namespace ILLIXR;
 
-static constexpr int   EYE_TEXTURE_WIDTH   = 1024;
-static constexpr int   EYE_TEXTURE_HEIGHT  = 1024;
+static constexpr int   EYE_TEXTURE_WIDTH   = DEFAULT_WIDTH;
+static constexpr int   EYE_TEXTURE_HEIGHT  = DEFAULT_HEIGHT;
```
Comment (e3m3) [inline gldemo/plugin.cpp:22]:
```suggestion
static constexpr int   EYE_TEXTURE_WIDTH   = ILLIXR::FB_WIDTH;
static constexpr int   EYE_TEXTURE_HEIGHT  = ILLIXR::FB_HEIGHT;
```

File: runtime/runtime_impl.hpp [hunk 6]
```diff
@@ -17,7 +18,7 @@ class runtime_impl : public runtime {
 		pb.register_impl<record_logger>(std::make_shared<sqlite_record_logger>());
 		pb.register_impl<gen_guid>(std::make_shared<gen_guid>());
 		pb.register_impl<switchboard>(create_switchboard(&pb));
-		pb.register_impl<xlib_gl_extended_window>(std::make_shared<xlib_gl_extended_window>(448*2, 320*2, appGLCtx));
+		pb.register_impl<xlib_gl_extended_window>(std::make_shared<xlib_gl_extended_window>(DEFAULT_WIDTH, DEFAULT_HEIGHT, appGLCtx));
```
Comment (e3m3) [inline runtime/runtime_impl.hpp:21]:
```suggestion
		pb.register_impl<xlib_gl_extended_window>(std::make_shared<xlib_gl_extended_window>(ILLIXR::FB_WIDTH, ILLIXR::FB_HEIGHT, appGLCtx));
```

File: timewarp_gl/plugin.cpp [hunk 7]
```diff
@@ -59,8 +60,8 @@ class timewarp_gl : public threadloop {
 	const std::shared_ptr<switchboard> sb;
 	const std::shared_ptr<pose_prediction> pp;
 
-	static constexpr int   SCREEN_WIDTH    = 550*2;
-	static constexpr int   SCREEN_HEIGHT   = 320*2;
+	static constexpr int   SCREEN_WIDTH    = DEFAULT_WIDTH;
+	static constexpr int   SCREEN_HEIGHT   = DEFAULT_HEIGHT;
```
Comment (e3m3) [inline timewarp_gl/plugin.cpp:64]:
```suggestion
	static constexpr int   SCREEN_WIDTH    = ILLIXR::FB_WIDTH;
	static constexpr int   SCREEN_HEIGHT   = ILLIXR::FB_HEIGHT;
```

**Review by e3m3 — CHANGES_REQUESTED** [review]
Sorry for the delay on this. We have upcoming changes in PR #198 that we now want your changes to match, so I've gone ahead an made the appropriate code suggestions that will make your changes compatible.

File: common/global_module_defs.hpp [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/*
+This is the file where default resolution values are defined
```
Comment (e3m3) [inline common/global_module_defs.hpp:2]:
```suggestion
This is the file where default values are defined
```

**Comment by DhairyaBahl** [comment]
@e3m3 Kindly Review it now !! I did everything as per your suggestions . Now it should work out :)

File: common/global_module_defs.hpp [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+#pragma once
+
+namespace ILLIXR{
+	
+#ifndef FB_WIDTH
+#define FB_WIDTH FB_WIDTH
+
+//Setting default Framebuffer width
+static constexpr int FB_WIDTH =2560; //Pixels
```
Comment (e3m3) [inline common/global_module_defs.hpp:13]:
```suggestion
static constexpr int FB_WIDTH = 2560; //Pixels
```

File: common/global_module_defs.hpp [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+#endif
+
+#ifndef FB_HEIGHT
+#define FB_HEIGHT FB_HEIGHT
+
+//Setting default framebuffer height
+static constexpr int FB_HEIGHT =1440; //Pixels
```
Comment (e3m3) [inline common/global_module_defs.hpp:20]:
```suggestion
static constexpr int FB_HEIGHT = 1440; //Pixels
```

File: gldemo/plugin.cpp [hunk 11]
```diff
@@ -14,11 +14,12 @@
...
 #include "common/gl_util/obj.hpp"
 #include "shaders/demo_shader.hpp"
+#include "common/global_module_defs.hpp"
 
 using namespace ILLIXR;
 
-static constexpr int   EYE_TEXTURE_WIDTH   = 1024;
-static constexpr int   EYE_TEXTURE_HEIGHT  = 1024;
+static constexpr int   EYE_TEXTURE_WIDTH   = ILLIXR::FB_WIDTH;
+static constexpr int   EYE_TEXTURE_HEIGHT  = ILLIXR:FB_HEIGHT;
```
Comment (e3m3) [inline gldemo/plugin.cpp:22]:
```suggestion
static constexpr int   EYE_TEXTURE_HEIGHT  = ILLIXR::FB_HEIGHT;
```

**Review by e3m3 — CHANGES_REQUESTED** [review]
Last few changes here

**Comment by DhairyaBahl** [comment]
@e3m3 I have made the suggested changes and squashed the commits. Kindly review the PR. :)

**Review by e3m3 — APPROVED** [review]

**Review by mhuzai — APPROVED** [review]

**Pull request merged**